common/analysis/lint_waiver.h

```diff
@@ -124,6 +136,13 @@ class LintWaiverBuilder {
   // TextStructureTokenized from text_structure_test_utils.h.
   void ProcessTokenRangesByLine(const TextStructureView&);
 
+  // Takes a set of active linter rules and the filename and the content of
+  // a waiver configuration file, returns true if applying the configuration
+  // succedded.
+  bool ApplyExternalWaivers(const std::set<absl::string_view> &active_rules,
```

Comment:
Consider having this return an `absl::Status` that can carry error messages/codes that can be surfaced to the caller, e.g. `kInvalidArgument`.  Then you can also test that things fail for the expected reason.

Reply:
How would the error message be handled? Should I display them first with VLOG and then use the same text in `absl::status`? As for the codes I think that most will use `kInvalidArgument` anyway.

Reply:
Each receiver of an `absl::Status` can decide who should see the error message, `std::cerr` or `VLOG(...) <<  status.message()`.  Or `return` it further up.  The generator of an `absl::Status` with an error doesn't have to print it.

Does that help?

Reply:
Should the return status contain the whole message?
Like:
`file:line:column: error: error message` etc?
Status seems like a good place to hold just the error message, that's why I asked if it should be both logged and returned as a status.

Additionally we will have to figure out the best way to store the concatenated string as absl::Status operates on a string_view as far as I can see.

Reply:
It can contain whatever is conveniently accessible at the time the error is detected.
The receivers can continue to build on the error messages.
`absl::Status` takes a `string_view` argument but internally stores a copy, so there are no lifetime issues.  
String copies are not a performance concern because these are on error handling conditions.

Reply:
Refactored everything to use `absl::Status`